spec/features/customers/show_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe "Customer Show Page" do
+  let!(:whole_foods) { Supermarket.create!(name: "Whole Foods", location: "San Francisco") }
+  let!(:safeway) { Supermarket.create!(name: "Safe Way", location: "Oakland") }
+
+  let!(:milk) { whole_foods.items.create!(name: "milk", price: 5.99 ) }
+  let!(:beef) { safeway.items.create!(name: "beef", price: 21.23 )}
+  let!(:cheese) { whole_foods.items.create!(name: "cheese", price: 12.99 ) }
+
+  let!(:tom) { Customer.create!(name: "Tom") }
+  let!(:jerry) { Customer.create!(name: "Jerry") }
+
+  let!(:customer_item_1) { CustomerItem.create!(customer_id: tom.id, item_id: milk.id) }
+  let!(:customer_item_2) { CustomerItem.create!(customer_id: tom.id, item_id: beef.id) }
+  let!(:customer_item_3) { CustomerItem.create!(customer_id: tom.id, item_id: cheese.id) }
+  let!(:customer_item_4) { CustomerItem.create!(customer_id: jerry.id, item_id: milk.id) }
+
+  describe "As a visitor" do
+    describe "When I visit customer show page" do
+      it "sees the customer's name, list of its items and the supermarket that it belongs to" do
+        visit "/customers/#{tom.id}"
+
+        expect(page).to have_content(tom.name)
+        within "item-#{milk.id}"
```

Comment:
To properly use a within block, youll need to add `do` and `end`, so I don't think these within blocks are actually being used. Just a heads up for next time!

Reply:
To make them work, you'll also want to add `#` at the start to indicate that you're referencing a CSS id selector (i.e. ` within "#item-#{milk.id}" do`